packages/auth/src/Providers/AmazonCognitoProvider.ts

```
@@ -15,29 +15,320 @@ import {
+		username: string,
+		password: string,
+		clientId: string,
+		clientMetaData: ClientMetaData,
+		signUpResult: AuthSignUpResult<UserAttributeKey>
+	) {
+		this.autoSignInInitiated = true;
+		const initiateAuthCommand = new InitiateAuthCommand({
+			AuthFlow: AuthFlowType.USER_PASSWORD_AUTH,
+			AuthParameters: {
+				username,
+				password
+			},
+			ClientId: clientId,
+			ClientMetadata: clientMetaData
+		});
+
+		if (signUpResult.isSignUpComplete) {
+			await this.signInAfterUserConfirmed(initiateAuthCommand, username);
+		} else if (this._config.signUpVerificationMethod === 'link') {
+			await this.handleLinkAutoSignIn(initiateAuthCommand, username);
+		} else {
+			await this.handleCodeAutoSignIn(initiateAuthCommand, username);
+		}
+
+	}
+
+	private async signInAfterUserConfirmed<UserAttributeKey extends AuthUserAttributeKey>(
+		initiateAuthCommand: InitiateAuthCommand,
+		username: string,
+		autoSignInPollingIntervalId?: ReturnType<typeof setInterval>
+	) {
+		try {
+			const result = await this.client.send(initiateAuthCommand);
+			let authSignInResult: AuthSignInResult<UserAttributeKey>;
+
+			if (result.AuthenticationResult) {
+				authSignInResult = {
+					isSignedIn: true,
+					nextStep: {
+						signInStep: AuthSignInStep.DONE
+					}
+				};
+			} else {
+				const challengeName = result.ChallengeName ?? '';
+				authSignInResult = {
+					isSignedIn: false,
+					nextStep: {
+						signInStep: this.mapChallengeNames(challengeName)
+					}
+				};
+			}
+
+			dispatchAuthEvent(
+				AuthHubEvent.AUTO_SIGN_IN,
+				{
+					response: authSignInResult,
+					user: null
+				},
+				`${username} has signed in successfully`
+			);
+
+			if (autoSignInPollingIntervalId) {
+				clearInterval(autoSignInPollingIntervalId);
+				this._storage.removeItem('amplify_polling_started');
+			}
+
+			this._storage.removeItem('amplify-auto-sign-in');
+		} catch (error) {
+			logger.error(error);
```

Is it intended to swallow the error?